### PR TITLE
Introduce and enforce linting rules

### LINKED ISSUE #44: Add pre commit hooks/ or github actions for linter checks

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR adds a basic and very minimal Eslint configuration.
Resolves #44

On top of that, it enforces these rules in a few different places:
- pre-commit hook
- GitHub action

### Current broken rules
Running `npm run lint` reveals 14 errors in the `src/` folder.
```
❯ npm run lint

> memestr@0.1.0 lint
> eslint --ext .js --max-warnings 0 src


/Users/metabase/code/btc-nostr/memestr/src/components/CategoryView/NFSWFeed.js
  143:33  error  'since' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/CategoryView/NatureProvider.js
  93:33  error  'since' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/HashtagTool/index.js
  105:33  error  'since' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/Login/index.js
  18:20  error  'props' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/LoginDropDownComponent/UploadUserPicture.js
  64:30  error  'event' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/Post/newPost.js
  158:24  error  Strings must use doublequote  quotes

/Users/metabase/code/btc-nostr/memestr/src/components/Post/post.js
  11:15  error  'props' is defined but never used  no-unused-vars
  28:35  error  'event' is defined but never used  no-unused-vars

/Users/metabase/code/btc-nostr/memestr/src/components/Posts/index.js
  15:17  error  Expected { after 'if' condition  curly

/Users/metabase/code/btc-nostr/memestr/src/components/Zap/index.js
   8:19  error  Strings must use doublequote  quotes
   8:55  error  Strings must use doublequote  quotes
  11:19  error  Strings must use doublequote  quotes
  11:55  error  Strings must use doublequote  quotes
  16:12  error  Empty block statement         no-empty

✖ 14 problems (14 errors, 0 warnings)
  6 errors and 0 warnings potentially fixable with the `--fix` option.
```

These have to be fixed in order for the CI to pass, which will also be a good test for the newly introduced Github Action.

### Results
1. As expected, [the initial GitHub Action run](https://github.com/AaruBama/memestr/actions/runs/6363547419/job/17279233962) failed.

image

2. After fixing linter errors (https://github.com/AaruBama/memestr/pull/54/commits/45e5eb382733d11dc78632ebe797bdff61001487)
3. [The next CI run](https://github.com/AaruBama/memestr/actions/runs/6364799459/job/17281323630) passed ✅

---
> **Important**
> I suggest turning on branch protection rules and making this check required. That will ensure no PR can be merged if it breaks the rules. Hence, `master` branch should always be in a good shape.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +62,7 @@
     "tailwindcss": "^3.3.3"
   },
   "lint-staged": {
-    "*.{md,yml,json,html,css,js,jsx,ts,tsx}": "prettier --write"
+    "*.{md,yml,json,html,css,js,jsx,ts,tsx}": "prettier --write",
+    "src/**/*.js": "npm run lint --fix"
```
Comment (nemanjaglumac) [inline package.json:66]: Apart from simply linting, attempt to fix errors in staged files.

File: src/components/Zap/index.js [hunk 2]
```diff
@@ -5,15 +5,14 @@ function setContentForMetadata(userDetails) {
     let content = null;
     let lnurl = null;
     if ("lud06" in userDetails && userDetails.lud06.length > 0) {
-        content = '{"lud06": "' + userDetails.lud06 + '"}';
+        content = JSON.stringify({ lud06: userDetails.lud06 });
         lnurl = userDetails.lud06;
     } else if ("lud16" in userDetails && userDetails.lud16.length > 0) {
-        content = '{"lud16": "' + userDetails.lud16 + '"}';
+        content = JSON.stringify({ lud16: userDetails.lud16 });
```
Comment (nemanjaglumac) [inline src/components/Zap/index.js:11]: If I understood correctly what you wanted to achieve here, the `content` needs to be JSON object.
I.e. `{"foo": "bar"}`

Better use `JSON.stringify()` than trying to construct it manually in that case.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "rules": {
+        "semi": ["error", "always"],
+        "quotes": ["error", "double"],
+        "no-unused-vars": ["error", { "vars": "all", "args": "all" }],
+        "curly": ["error", "all"]
+    },
+    "extends": ["eslint:recommended", "react-app"]
+}
```
Comment (nemanjaglumac) [inline .eslintrc:9]: This list can absolutely be longer, but it is more than sufficient for this project at this stage in development.
The point is not to mail down every single rule. Just start with some sane defaults and you'll update the rules later on (if needed).

File: .github/workflows/linter.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Linter
+
+on:
+  pull_request:
```
Comment (nemanjaglumac) [inline .github/workflows/linter.yml:4]: I know that you originally wanted to run the action only on PRs against the master branch, but I highly suggest running it for all PRs.

You can tweak and optimize later, if needed.

File: .eslintrc (hunk 3 above)
Comment (AaruBama) [reply .eslintrc:9]: +1

File: src/components/Zap/index.js (hunk 2 above)
Comment (AaruBama) [reply src/components/Zap/index.js:11]: This is a Deprecated file. I will remove it in the next PR.

**Comment by nemanjaglumac** [comment]
Had to force-push to resolve merge-conflicts so the action will run again. And it's passing.

You somehow ended up with TWO `devDependencies` after you merged #53 and #49.

File: package.json [hunk 5]
```diff
@@ -27,17 +27,7 @@
...
-    "lint-staged": "^14.0.1",
-    "prettier": "^3.0.3",
-    "prettier-plugin-tailwindcss": "^0.5.3",
-    "source-map-loader": "^4.0.1",
-    "tailwindcss": "^3.3.3"
-  },
-  "devDependencies": {
-    "@svgr/cli": "^8.1.0",
-    "@svgr/webpack": "^8.1.0",
+    "eslint": "^8.50.0",
```
Comment (nemanjaglumac) [inline package.json:30]: The only actual change was adding/installing `eslint` here, but there were two `devDependencies` in `master` that I had to resolve.

File: .eslintrc (hunk 3 above)
Comment (AaruBama) [reply .eslintrc:9]: should we be throwing a `warning` instead of `error` on curly?

**Pull request merged**